### DISCUSSION TITLE
Best practice for reconnecting after retryAttempts has been passed?

### DISCUSSION THREAD

**Discussion by morajabi** [enisdenjo/graphql-ws#121] [Q&A] [answered]
In our case, the computer is sleep for 8 hours, and then starts up, now I need to reconnect the socket. What do you recommend? 

What I have in mind is: recreate using `createWsClient` and therefore creating a new `urql` client (I use urql) using the new Graphql WS client. But this seems to be inefficient.

**Comment by enisdenjo** [comment, marked as answer]
[Check out the client usage with custom retry timeout strategy recipe.](https://github.com/enisdenjo/graphql-ws#retry-strategy) You can time your retries exactly to your needs by controlling the [`retryWait`](https://github.com/enisdenjo/graphql-ws/blob/master/docs/interfaces/client.clientoptions.md#retrywait) promise resolution.

After the `retryAttempts` have passed, the client will report an error.

**Reply by morajabi** [reply]
That makes sense. But if I understand correctly I'll have to pick large `retryAttempts` just to be safe in this case, do you think that's ok? As I don't want to ever be disconnected from the server no matter how long the app is open.

Update:
Added this:
```
export async function waitForOnline(): Promise<void> {
  return new Promise(resolve => {
    function now() {
      window.removeEventListener('online', now)
      resolve()
    }
    window.addEventListener('online', now)
  })
}
```

and 
```
 retryWait: async function randomisedExponentialBackoff(retries: number) {
      if (!navigator.onLine && retries > 5) {
        await waitForOnline()
      }
      // ...
```

**Reply by enisdenjo** [reply]
You can just do `retryAttempts: Infinity` to never ever stop retrying.
